NURESTConnection.js

```diff
@@ -1,6 +1,116 @@
 import NUObject from './NUObject';
 import NUInterceptor from './NUInterceptor';
 import { getLogger } from './Logger';
+import { Enum } from 'enumify';
+
+class ResponseCodeEnum extends Enum {
+    static getClassName() {
+        return 'ResponseCodeEnum';
+    }
+
+    static getEnumForCode (code) {
+        return ResponseCodeEnum.enumValues.filter( item => item.code === code);
+    }
+
+    getClassName() {
+        return ResponseCodeEnum.getClassName();
+    }
+
+    static getErrors(result) {
+        const { response } = result;
+        switch (response.status) {
+            case 400:
+            case 401:
+            case 403:
+            case 404:
+            case 405:
+            case 500:
```

Comment:
why not handle 409, 412, and 503 here? we have enums defined for these 3 as well